src/OpenTelemetry.Api/Context/CorrelationContext.cs

```diff
@@ -62,23 +69,62 @@ internal CorrelationContext(List<CorrelationContextEntry> entries)
         public static bool operator !=(CorrelationContext left, CorrelationContext right) => !(left == right);
 
         /// <summary>
-        /// Gets the <see cref="CorrelationContextEntry"/> with the specified name.
+        /// Retrieves a correlation item.
+        /// </summary>
+        /// <param name="key">Correlation item key.</param>
+        /// <returns>Retrieved correlation value or <see langword="null"/> if no match was found.</returns>
+        public string GetCorrelation(string key)
+            => this.activity?.GetBaggageItem(key);
+
+        /// <summary>
+        /// Adds a correlation item.
+        /// </summary>
+        /// <param name="key">Correlation item key.</param>
+        /// <param name="value">Correlation item value.</param>
+        /// <returns>The <see cref="CorrelationContext"/> instance for chaining.</returns>
+        public CorrelationContext AddCorrelation(string key, string value)
```

Comment:
This API will not match the spec, right? As spec wants "Set" behavior, but we have "add" behavior, and also we won't have "Remove" feature.

Reply:
Correct. The current version is built on top of Activity.Baggage, which is very limiting. Let's discuss this. The other option is we could make CorrelationContext independent of Baggage which would allow us to fully implement the spec. If we do that, let's remove "AllChildren" from #969. CorrelationContext could be used for "AllChildren" behavior and then EnrichmentScope is just responsible for "FirstChild" behavior?